homeassistant/util/yaml.py

```diff
@@ -39,7 +39,10 @@ def _ordered_dict(loader, node):
     nodes = loader.construct_pairs(node)
     dups = [k for k, v in Counter(k for k, _ in nodes).items() if v > 1]
     if dups:
-        raise yaml.YAMLError("ERROR: duplicate keys: {}".format(dups))
+        msg = repr_helper(nodes)
```

Comment:
We get nicer output from the repr_helper if we first convert it to a dict. We could also cut the message off here.
```python
msg = repr_helper(dict(nodes))[0:100]
```

We will however lose the duplicate keys if we cast to dict.